File: AddressBook/AddressBook/Dtos/PersonWebsite.cs
using AddressBook.Entities;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using AddressBook.Common.Paging;

namespace AddressBook.Dtos
{
    public class PersonWebsiteRes
    {
        public int PersonWebsiteId { get; set; }
        public string? Website { get; set; }
        public int? PersonId { get; set; }
        public PersonRes? Person { get; set; }
        public int? WebsiteLabelId { get; set; }
        public WebsiteLabelRes? WebsiteLabel { get; set; }
    }

    public class PersonWebsiteReqEdit
    {
        [Required]
        public string? Website { get; set; }
        public int? PersonId { get; set; }
        public int? WebsiteLabelId { get; set; }
    }

    public class PersonWebsiteReqSearch : PagedReq
    {
        [Required]
        public int? PersonId { get; set; }
    }
}

File: AddressBook/react-client/src/dtos/PersonWebsite.ts
import Common from "../utility/Common";
import { PagedReq } from "./PagedReq";
import { PersonRes } from "./Person";
import { WebsiteLabelRes } from "./WebsiteLabel";

export interface PersonWebsiteRes {
  personWebsiteId?: number;
  website?: string;
  personId?: number;
  person?: PersonRes;
  websiteLabelId?: number;
  websiteLabel?: WebsiteLabelRes;
}

export class PersonWebsiteReqEdit {
  website?: string = "";
  personId?: number = 0;
  websiteLabelId?: number = 0;
}

export class PersonWebsiteReqSearch extends PagedReq {
  personId?: string;
  constructor(
    {
      pageNumber = 1,
      pageSize = Common.DEFAULT_PAGE_SIZE,
      orderBy,
      searchText = "",
    }: PagedReq,
    {personId = ""}
  ) {
    super({
      pageNumber: pageNumber,
      pageSize: pageSize,
      orderBy: orderBy,
      searchText: searchText,
    });
    this.personId = personId;
  }
}